index.html

```diff
@@ -5113,48 +5131,47 @@ <h3 id="playback-AudioBufferSourceNode">
-                            <code>true</code>.
+                            <li>If <code>offset</code> &lt;
+                            <code>actualLoopEnd</code> and
+                            <code>bufferTime</code> &gt;=
+                            <code>actualLoopStart</code>, set
+                            <code>enteredLoop</code> to <code>true</code> since
+                            playback began before or within the loop, and the
+                            playhead is now past the start of the loop.
                             </li>
-                          </ol>
-                        </li>
-                        <li>Let <code>effectiveBufferTime</code> equal
-                        <code>bufferTime</code>.
-                        </li>
-                        <li>If <code>loop</code> is true, map the playhead
-                        position to the correct point within the loop body
-                        accounting for loop iterations:
-                          <ol>
-                            <li>While <code>effectiveBufferTime</code> &gt;=
-                            <code>actualLoopEnd</code>,
-                              <ol>
-                                <li>Let <code>effectiveBufferTime</code> equal
-                                <code>effectiveBufferTime - (actualLoopEnd -
-                                actualLoopStart)</code>.
-                                </li>
-                              </ol>
+                            <li>If <code>offset</code> &gt;=
+                            <code>actualLoopEnd</code> and
+                            <code>bufferTime</code> &lt;
+                            <code>actualLoopEnd</code>, set
+                            <code>enteredLoop</code> to <code>true</code> since
+                            playback began after the loop, and the playhead is
+                            now before the end of the loop.
                             </li>
-                            <li>While <code>enteredLoop</code> is true and
-                            <code>effectiveBufferTime</code> &lt;
-                            <code>actualLoopStart</code>,
+                            <li>If <code>enteredLoop</code> is true, map the
+                            playhead position to the correct point within the
+                            loop body accounting for loop iterations:
                               <ol>
-                                <li>Let <code>effectiveBufferTime</code> equal
-                                <code>effectiveBufferTime + (actualLoopEnd -
-                                actualLoopStart)</code>.
+                                <li>While <code>bufferTime</code> &gt;= <code>
+                                  actualLoopEnd</code>, let
+                                  <code>bufferTime</code> equal
+                                  <code>bufferTime - (actualLoopEnd -
+                                  actualLoopStart)</code>.
```

Comment:
If loop duration is very short and playbackRate is huge, doing one subtraction might not be enough to wrap bufferTime back into the loop area.  Same issue for the case below.